api/tests/mocha/services/connected-user-log.spec.js

```diff
@@ -0,0 +1,35 @@
+const chai = require('chai');
+const sinon = require('sinon');
+const connectedUserLogService = require('../../../src/services/connected-user-log');
+const db = require('../../../src/db');
+
+let clock;
+
+describe('Connected Users Log service', () => {
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+  });
+
+  afterEach(() => {
+    sinon.restore();
+    clock.restore();
+  });
+
+  describe('saveLog()', () => {
+    it('should save a log', () => {
+      sinon.stub(db.medicLogs, 'get').rejects({ status: 404 });
+      const putStub = sinon.stub(db.medicLogs, 'put').resolves();
+      const expectedDoc = {
+        _id: 'connected-user-userXYZ',
+        user: 'userXYZ',
+        timestamp: 0
+      };
+
+      return connectedUserLogService
+        .save('userXYZ')
+        .then(() => {
+          chai.expect(putStub.args[0][0]).to.deep.include(expectedDoc);
+        });
+    });
```

Comment:
Remember to unit test the negatives! I think there are at least 4 more tests we want here...

1. What happens if the `put` call rejects?
2. What happens if the `get` call rejects?
3. What about if the `get` call returns an existing doc that's within the 7 day limit?
4. What about if the `get` call returns an existing doc that's older than the 7 day limit?

Anything else you can think of?

Reply:
Good catch. I have added more unit tests to test the negatives as well.